sdk/search/ai-search/README.md

```diff
@@ -0,0 +1,302 @@
+  console.log(result);
+}
+```
+
+For a more advanced search that uses [Lucene syntax](https://docs.microsoft.com/en-us/azure/search/query-lucene-syntax), specify `queryType` to be `all`:
+
+```js
+for await (const result of client.listSearchResults('category:budget AND "recently renovated"^3', {
+  queryType: "full",
+  searchMode: "all"
+})) {
+  console.log(result);
+}
+```
+
+### Querying with OData filters
+
+Using the `filter` query parameter allows you to query an index using the syntax of an [OData \$filter expression](https://docs.microsoft.com/en-us/azure/search/search-query-odata-filter).
+
+```js
+const { odata } = require("@azure/ai-search");
+
+let baseRateMax = 200;
+let ratingMin = 4;
+for await (const result of client.listSearchResults("WiFi", {
+  filter: odata`Rooms/any(room: room/BaseRate lt ${baseRateMax}) and Rating ge ${ratingMin}`,
+  orderBy: "Rating desc"
+  select: ["HotelId", "HotelName", "Rating"]
+})) {
+  // Each result will have "HotelId", "HotelName", and "Rating"
+  // in addition to the standard search result property "score"
+  console.log(result);
+}
+```
+
+### Querying with facets
+
+[Facets](https://docs.microsoft.com/en-us/azure/search/search-filters-facets) are used to help a user of your application refine a search along pre-configured dimensions. [Facet syntax](https://docs.microsoft.com/en-us/rest/api/searchservice/search-documents#facetstring-zero-or-more) provides the options to sort and bucket facet values.
+
+```js
+const page = (
+  await client
+    .listSearchResults("WiFi", {
+      facets: ["Category,count:3,sort:count", "Rooms/BaseRate,interval:100"]
+    })
+    .byPage()
+    .next()
+).value;
+
+// {
+//   Rating: [
+//     { count: 4, value: 2 },
+//     { count: 5, value: 3 },
+//     { count: 8, value: 4 }
+//   ],
+//   'Rooms/BaseRate': [
+//     { count: 16, value: 0 },
+//     { count: 17, value: 100 },
+//     { count: 17, value: 200 }
+//   ],
+//   Category: [
+//     { count: 5, value: 'Budget' },
+//     { count: 5, value: 'Luxury' },
+//     { count: 5, value: 'Resort and Spa' }
+//   ]
+// }
+console.log(page.facets);
+```
+
+When retrieving results, an `facets` property will be available that will indicate the number of results that fall into each facet bucket. This can be used to drive refinement (e.g. issuing a follow-up search that filters on the `Rating` being greater than 3 and less than 4.)
+
+### Retrieving documents by id
+
+A specific document can be retrieved by its primary key value:
+
+```js
+const result = await client.getDocument("1234");
+console.log(result);
+```
+
+### Retrieve suggestions from an index
+
+If you [created a suggester](https://docs.microsoft.com/en-us/azure/search/index-add-suggesters) on your index, you can use it to return result suggestions for a user query.
+
+This example shows returning the top three suggestions for the input "wifi" from the suggester "sg":
+
+```js
+const suggestResult = await client.suggest("sg", "wifi", {
```

Comment:
API question - if you have to specify the object literal anyways would it be nice to just have the `suggester` and `input` be fields in there rather than being positional?

Or are the other examples optional and unlikely to be used?

Reply:
In the raw shape they are on the options bag, but they're not actually optional, so I promoted them to be required parameters. Everything left in the options bag *is* optional.

Would you prefer everything be in one input object?

Reply:
So between "sg" and "wifi" - is there an intuitive order between them? Will it be easy to transpose them (on accident)? Will their values often be obvious enough that if you're looking at the call site you know which is which?

I do agree with your logic in general and I don't honestly know the specifics of this API well enough to answer my own questions :)

Reply:
It looks like the track1 .NET lib did `(searchText, suggesterName, options)` -- I'm not sure if one way is intrinsically better than the other as both are required.

I get that it's easy to transpose arguments of the same type, but that's true of any method that takes multiple inputs of the same type, no? At some point you need IntelliSense/docs to keep things straight.

It's a good point that maybe we should just embrace the object inputs for clarity...

Reply:
After some thought (and talking to others) I am moving towards only using the object form for this as you suggested. Partly this is because I realized that `searchText` is optional on actual search operations (it defaults to *) and it felt weird to pull it out for suggest/autocomplete but not search.